docs/spec/v2alpha1/helmreleases.md

```diff
@@ -0,0 +1,276 @@
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Wait tells the reconciler to wait with marking a Helm action as
+	// successful until all resources are in a ready state.
+	// If set, it will wait for as long as 'Timeout'.
+	// +optional
+	Wait bool `json:"wait,omitempty"`
+
+	// MaxHistory is the number of revisions saved by Helm for this release.
+	// Use '0' for an unlimited number of revisions; defaults to '10'.
+	// +optional
+	MaxHistory *int `json:"maxHistory,omitempty"`
+
+	// Install holds the configuration for performing Helm install actions.
+	// +optional
+	Install *struct {
+		// Timeout for the Helm install action.
+		// Defaults to the main 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// Wait tells the reconciler to wait with marking the Helm install as
+		// successful until all resources are in a ready state. When set, it will
+		// wait for as long as 'Timeout'. Defaults to the main 'Wait' value.
+		// +optional
+		Wait bool `json:"wait,omitempty"`
+
+		// SkipCRDs will mark this Helm release to skip the creation of CRDs during
+		// the installation.
+		// +optional
+		SkipCRDs bool `json:"skipCRDs,omitempty"`
+	} `json:"install,omitempty"`
+	
+	// Upgrade holds the configuration for performing Helm upgrade actions.
+	// +optional
+	Upgrade *struct {
+		// Timeout for the Helm upgrade action. Defaults to the main
+		// 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// Wait tells the reconciler to wait with marking the Helm upgrade as
+		// successful until all resources are in a ready state. When set, it will
+		// wait for as long as 'Timeout'. Defaults to the main 'Wait' value.
+		// +optional
+		Wait bool `json:"wait,omitempty"`
+
+		// ResetValues resets all values to the chart defaults, so that the
+		// configuration applied is what is defined in the resource values.
+		// Defaults to true to ensure declarative behaviour.
+		// +optional
+		ResetValues *bool `json:"resetValues,omitempty"`
+	} `json:"upgrade,omitempty"`
+
+	// Test holds the configuration for enabling and performing Helm
+	// test actions.
+	// +optional
+	Test *struct {
+		// Enable tells the reconciler to perform a test action
+		// on the configured condition reasons. Defaults to false.
+		// +optional
+		Enable bool `json:"enable,omitempty"`
+
+		// OnConditionReason defines on what condition reasons a Helm test
+		// should be performed. Defaults to ['InstallSucceeded', 'UpgradeSucceeded'].
+		// +optional
+		OnConditionReason []string `json:"onConditionReason,omitempty"`
+
+		// Timeout for the Helm test action. Defaults to the main
+		// 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+	} `json:"test,omitempty"`
+
+	// Rollback holds the configuration for enabling and performing Helm
+	// rollback actions.
+	// +optional
+	Rollback *struct {
```

Comment:
A similar `Uninstall` key would be useful as well, see https://github.com/fluxcd/helm-operator/issues/388. In addition to what's mentioned there, an `Uninstall.OnConditionReason` would be needed in order to ignore test failures after installs.

Reply:
This is planned, see developments in #2. This POC also no longer relies on condition reasons but on condition types and statuses.